src/controllers/csv-processor.ts

```diff
@@ -250,3 +315,96 @@ export const processCSV = async (dataset: Dataset, page: number, size: number):
         };
     }
 };
+
+export const processCSVFromBlobStorage = async (
+    dataset: Dataset,
+    page: number,
+    size: number
+): Promise<ViewErrDTO | ViewDTO> => {
+    const blobStoageService = new BlobStorageService();
+    try {
+        const datafiles = await dataset.datafiles;
+        const datafile: Datafile | undefined = datafiles
+            .filter((filterfile: Datafile) => filterfile.draft === true)
+            .sort(
+                (first: Datafile, second: Datafile) =>
+                    new Date(second.creationDate).getTime() - new Date(first.creationDate).getTime()
+            )
+            .shift();
+        if (datafile === undefined || datafile === null) {
+            return {
+                success: false,
+                errors: [
+                    {
+                        field: 'dataset',
+                        message: [
+                            { lang: ENGLISH, message: t('errors.no_datafile', { lng: ENGLISH }) },
+                            { lang: WELSH, message: t('errors.no_datafile', { lng: WELSH }) }
+                        ],
+                        tag: { name: 'erorors.no_datafile', params: {} }
+                    }
+                ],
+                dataset_id: dataset.id
+            };
+        }
+        const buff = await blobStoageService.readFile(`${datafile.id}.csv`);
+
+        const dataArray: Array<Array<string>> = (await parse(buff, {
+            delimiter: ','
+        }).toArray()) as string[][];
+        const csvheaders = dataArray.shift();
+        const total_pages = Math.ceil(dataArray.length / size);
```

Comment:
This has the same issue as a previous commit where we read in the whole file only to throw most of it away. So a paginated file amplifies the load on the service by the number of pages instead of reducing it by the number of pages.
If we need this information then we can could cache it in a rows_hint field in the database. We'd probably also need some index data that told us the offset in bytes of each page.

Reply:
Except pages aren't fixed in size... They can be any number from 5 to 500.  The current list is: 

- 5
- 10
- 20
- 50
- 100
- 250
- 500

We will have to paginate, we will have to paginate dynamically as well based on the users choices.  If its a big file we can't render everything directly to the frontend we will have to provide means to paginate that data until we are able to get it into the online cube.  Once its in the online cube we can query a specific number of lines and we can query a specific starting point.  CSV sadly doesn't give us the ability.